components/eam/src/chemistry/mozart/chemistry.F90

```diff
@@ -1516,10 +1540,84 @@ subroutine chem_timestep_tend( state, ptend, cam_in, cam_out, dt, pbuf,  fh2o, f
                 !end if
 
              else
-                Diff(:ncol,1) = (ftem(:ncol,1) - gas_ac_2D(:ncol))/dt
+                Diff(:ncol,1) = (ftem_layers(:ncol,1) - gas_ac_2D(:ncol))/dt
              end if
              call outfld(trim(solsym(n))//'_2DTDO', Diff(:ncol,1), pcols, lchnk )
            end if
+           ! HHLEE 20210923
+           if (history_gaschmbudget_2D_levels ) then
+             ftem_layers = 0.0_r8
+             gas_ac_layers = 0.0_r8
+
+             gas_ac_idx = pbuf_get_index(gas_ac_name(n))
+             call pbuf_get_field(pbuf, gas_ac_idx, gas_ac )
+
+             if (gaschmbudget_2D_L1_e .lt. gaschmbudget_2D_L1_s .or. gaschmbudget_2D_L2_e .lt. gaschmbudget_2D_L2_s .or. &
+                 gaschmbudget_2D_L3_e .lt. gaschmbudget_2D_L3_s .or. gaschmbudget_2D_L4_e .lt. gaschmbudget_2D_L4_s ) then
+                   call endrun('chem_readnl: ERROR 2D chem diags layers')
```

Comment:
I'd suggest we provide more informative error messages, like 'chem_readnl: ERROR 2D chem diags layers, layer ending index is less than starting index'.

Reply:
Changed.